### PR TITLE
ST-490 Play upgrade 2.5.19 - part 1

### PR DESCRIPTION
**Pull request by lewey**
- [x] Unit tests passing
- [x] Coverage reached
- [x] Acceptance tests passing
- [ ] Reviewed

### CODE REVIEW

File: app/uk/gov/hmrc/tai/model/TaxYear.scala [hunk 1]
```diff
@@ -37,8 +37,13 @@ case class TaxYear(year: Int) extends Ordered[TaxYear] {
   def compare(that: TaxYear) = this.year compare that.year
   def twoDigitRange = s"${start.year.get % 100}-${end.year.get % 100}"
   def fourDigitRange = s"${start.year.get}-${end.year.get}"
+  def within(currentDate: LocalDate): Boolean = {
+    (currentDate.isEqual(TaxYear().start) || currentDate.isAfter(TaxYear().start)) &&
```
Comment (lewey) [inline app/uk/gov/hmrc/tai/model/TaxYear.scala]: No need for TaxYear() here

File: test/controllers/income/bbsi/BbsiCloseAccountControllerSpec.scala [hunk 2]
```diff
@@ -141,7 +140,7 @@ class BbsiCloseAccountControllerSpec extends PlaySpec
 
         val sut = createSUT
 
-        val date = TaxYearResolver.startOfCurrentTaxYear.minusDays(1)
+        val date = TaxYear().endPrev
```
Comment (oc-ste) [inline test/controllers/income/bbsi/BbsiCloseAccountControllerSpec.scala]: Is this the same as TaxYearResolver.startOfCurrentTaxYear.minusDays(1)?

File: test/controllers/income/bbsi/BbsiCloseAccountControllerSpec.scala [hunk 3]
```diff
@@ -189,7 +188,7 @@ class BbsiCloseAccountControllerSpec extends PlaySpec
 
         val sut = createSUT
 
-        val date = TaxYearResolver.endOfCurrentTaxYear.plusDays(1)
+        val date = TaxYear().next.start
```
Comment (oc-ste) [inline test/controllers/income/bbsi/BbsiCloseAccountControllerSpec.scala]: Is this the same as TaxYear().end?

File: test/uk/gov/hmrc/tai/service/AuditServiceSpec.scala [hunk 4]
```diff
@@ -493,8 +493,8 @@ class AuditServiceSpec extends PlaySpec with MockitoSugar with FakeTaiPlayApplic
       tags = Map("clientIP" -> trueClientIp.getOrElse("-"),
         "path" -> path.getOrElse("NA"), "X-Session-ID" -> sessionId.getOrElse("-"),
         "Akamai-Reputation" -> "-", "X-Request-ID" -> requestId.getOrElse("-"),
-        "deviceID" -> deviceId.getOrElse("-"), "ipAddress" -> ipAddress.getOrElse("-"),
-        "token" -> "-", "clientPort" -> clientPort.getOrElse("-"), "Authorization" -> authorization.getOrElse("-"),
+        "deviceID" -> deviceId.getOrElse("-"),
+        "clientPort" -> clientPort.getOrElse("-"),
         "transactionName" -> auditType), detail = detail
```
Comment (oc-ste) [inline test/uk/gov/hmrc/tai/service/AuditServiceSpec.scala]: is ipAddress, token and authorization suppose to go here?

File: project/plugins.sbt [hunk 5]
```diff
@@ -6,7 +6,7 @@ resolvers ++= Seq(
 
 resolvers += "HMRC Releases" at "https://dl.bintray.com/hmrc/releases"
 
-addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.5.12")
+addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.5.19")
```
Comment (oc-ste) [inline project/plugins.sbt]: Suppose to upgrade this?

File: test/uk/gov/hmrc/tai/service/TaxPeriodLabelServiceSpec.scala [hunk 6]
```diff
@@ -21,15 +21,15 @@ import org.scalatestplus.play.PlaySpec
...
 
   "TaxPeriodLabelService " should {
 
     "generate tax period label" in {
-      TaxPeriodLabelService.taxPeriodLabel(2017) mustBe HtmlFormatter.htmlNonBroken("6 April 2017") + " to " + HtmlFormatter.htmlNonBroken("5 April 2018")
-      TaxPeriodLabelService.taxPeriodLabel(2016) mustBe HtmlFormatter.htmlNonBroken("6 April 2016") + " to " + HtmlFormatter.htmlNonBroken("5 April 2017")
+      TaxPeriodLabelService.taxPeriodLabel(2017) mustBe s"${HtmlFormatter.htmlNonBroken("6 April 2017")} " +
+        s"${messagesApi("language.to")} ${HtmlFormatter.htmlNonBroken("5 April 2018")}"
+      TaxPeriodLabelService.taxPeriodLabel(2016) mustBe s"${HtmlFormatter.htmlNonBroken("6 April 2016")} " +
+        s"${messagesApi("language.to")} ${HtmlFormatter.htmlNonBroken("5 April 2017")}"
```
Comment (oc-ste) [inline test/uk/gov/hmrc/tai/service/TaxPeriodLabelServiceSpec.scala]: Is behaviour change expected here?

File: test/uk/gov/hmrc/tai/viewModels/income/BbsiClosedCheckYourAnswersViewModelSpec.scala [hunk 7]
```diff
@@ -95,6 +95,6 @@ class BbsiClosedCheckYourAnswersViewModelSpec extends PlaySpec
     }
   }
 
-  val dateBeforeCurrentTaxYear = TaxYearResolver.startOfCurrentTaxYear.minusDays(1)
-  val dateWithinCurrentTaxYear = TaxYearResolver.startOfCurrentTaxYear
+  val dateBeforeCurrentTaxYear = TaxYear().endPrev
```
Comment (oc-ste) [inline test/uk/gov/hmrc/tai/viewModels/income/BbsiClosedCheckYourAnswersViewModelSpec.scala]: Is this the same as:
TaxYearResolver.startOfCurrentTaxYear.minusDays(1)

File: test/views/html/incomes/BankBuildingSocietyCheckYourAnswersSpec.scala [hunk 8]
```diff
@@ -45,7 +46,7 @@ class BankBuildingSocietyCheckYourAnswersSpec extends TaiViewSpec {
       "links to the BBSI end date form page" when {
         "the account is closed before the current tax year" in {
 
-          val date = TaxYearResolver.startOfCurrentTaxYear.minusDays(1)
+          val date = TaxYear().endPrev
```
Comment (oc-ste) [inline test/views/html/incomes/BankBuildingSocietyCheckYourAnswersSpec.scala]: Same as:TaxYearResolver.startOfCurrentTaxYear.minusDays(1)?

File: app/uk/gov/hmrc/tai/model/TaxYear.scala [hunk 9]
```diff
@@ -37,8 +37,13 @@ case class TaxYear(year: Int) extends Ordered[TaxYear] {
...
+      (currentDate.isBefore(TaxYear().end) || currentDate.isEqual(TaxYear().end))
+  }
```
Comment (oc-ste) [inline app/uk/gov/hmrc/tai/model/TaxYear.scala]: dedupe

File: test/controllers/income/bbsi/BbsiCloseAccountControllerSpec.scala (hunk 2 above)
Comment (lewey) [reply test/controllers/income/bbsi/BbsiCloseAccountControllerSpec.scala]: Or TaxYear().prev.end

File: test/controllers/income/bbsi/BbsiCloseAccountControllerSpec.scala (hunk 3 above)
Comment (lewey) [reply test/controllers/income/bbsi/BbsiCloseAccountControllerSpec.scala]: TaxYearResolver.endOfCurrentTaxYear is the same as TaxYear().end

File: project/plugins.sbt (hunk 5 above)
Comment (lewey) [reply project/plugins.sbt]: changed back

File: test/uk/gov/hmrc/tai/service/TaxPeriodLabelServiceSpec.scala (hunk 6 above)
Comment (lewey) [reply test/uk/gov/hmrc/tai/service/TaxPeriodLabelServiceSpec.scala]: Issue with htmlNonBroken related tests failing due to spacing. The above approach was the fix.

File: test/uk/gov/hmrc/tai/viewModels/income/BbsiClosedCheckYourAnswersViewModelSpec.scala (hunk 7 above)
Comment (lewey) [reply test/uk/gov/hmrc/tai/viewModels/income/BbsiClosedCheckYourAnswersViewModelSpec.scala]: Yes

File: test/views/html/incomes/BankBuildingSocietyCheckYourAnswersSpec.scala (hunk 8 above)
Comment (lewey) [reply test/views/html/incomes/BankBuildingSocietyCheckYourAnswersSpec.scala]: Yes

File: app/uk/gov/hmrc/tai/model/TaxYear.scala (hunk 9 above)
Comment (lewey) [reply app/uk/gov/hmrc/tai/model/TaxYear.scala]: Made a call directly to the methods

**Pull request merged**